### PR TITLE
TE7-T893: Add UI Components Assets2

### PR DESCRIPTION
**Pull request by Kruna1Pate1**
UI Components Assets2

### Summary by AIReviewRadar 1.0.2

Release Notes:

- New Feature: Added several UI components such as circular progress bars, gradient buttons, and rounded rectangles to improve the visual design of the app's user interface.

> "New UI components galore,
> Circular progress bars and more.
> Gradient buttons, rounded shapes,
> Our app's design now truly takes shape!"

### CODE REVIEW

File: Demo/app/src/main/java/com/krunal/demo/DemoApplication.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.krunal.demo
+
+import android.app.Application
+import com.krunal.demo.uicomponents.helpers.PreferenceHelper
+
+class DemoApplication: Application() {
```
Comment (nishchal-v) [inline Demo/app/src/main/java/com/krunal/demo/DemoApplication.kt]: We should not add these code changes as this is a Assets PR

File: Demo/app/build.gradle [hunk 2]
```diff
@@ -32,13 +33,25 @@ android {
...
 
 dependencies {
-
+    def lifecycle_version = "2.6.1"
     implementation 'androidx.core:core-ktx:1.10.1'
     implementation 'androidx.appcompat:appcompat:1.6.1'
     implementation 'com.google.android.material:material:1.9.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$lifecycle_version"
+    implementation 'androidx.fragment:fragment-ktx:1.5.7'
```
Comment (nishchal-v) [inline Demo/app/build.gradle]: Don't commit changes in this PR. Upgrades should be done in separate PR

Comment (nishchal-v) [reply Demo/app/build.gradle]: Is this resolved?

File: Demo/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -3,14 +3,24 @@
     xmlns:tools="http://schemas.android.com/tools">
 
     <application
+        android:name=".DemoApplication"
```
Comment (nishchal-v) [inline Demo/app/src/main/AndroidManifest.xml]: Would this still work?

File: Demo/app/build.gradle (hunk 2 above)
Comment (Kruna1Pate1) [reply Demo/app/build.gradle]: Yes moved this to helper PR
